genext2fs.c

```diff
@@ -3362,16 +3486,20 @@ main(int argc, char **argv)
 		stats.ninodes = EXT2_FIRST_INO - 1 + (nbresrvd ? 1 : 0);
 		stats.nblocks = 0;
 
-		populate_fs(NULL, dopt, didx, squash_uids, squash_perms, fs_timestamp, &stats);
+		if (nbinodes == 0) {
```

Comment:
This feature should be in its own merge request.

Reply:
I don't disagree. But what other way do you see to compute the number of inodes for when the tarball comes from stdin? In that case, we cannot call `populate_fs` twice to figure out the required number of inodes in the first call. One way would be to require the `-i` option (bytes per inode) being set to auto-compute the value from the given number of blocks.

Reply:
`-t -` (or even `-D -`) would imply a mode without precomputing inodes number, and should require `-N 0` or something else specifying this number. But as I said, it's another feature for another merge request.